### PR TITLE
Updates README.md and removes outdated demo

### PR DESCRIPTION
**Pull request by JoaoBraveCoding**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,17 +35,63 @@ The logging-ocm-addon consists of one component:
...
+          openTelemetryCollector:
+            enabled: false
+  observabilityAddonSpec: {}
+  storageConfig:
+    metricObjectStorage:
+      name: thanos-object-storage
+      key: thanos.yaml
+```
+
+Note: Deploy a custom image by adding the annotation: `mco-multicluster_observability_addon-image: quay.io/YOUR_ORG_HERE/multicluster-observability-addon:YOUR_TAG_HERE`
```
Comment (xperimental) [inline README.md:81]: The "counting enumeration" only continues when you indent this with 4 spaces:

```suggestion
    ```yaml
    apiVersion: observability.open-cluster-management.io/v1beta2
    kind: MultiClusterObservability
    metadata:
      name: observability
    spec:
      capabilities:
        platform:
          logs:
            collection:
              enabled: true
        userWorkloads:
          logs:
            collection:
              clusterLogForwarder:
                enabled: true
          traces:
            collection:
              instrumentation:
                enabled: true
              openTelemetryCollector:
                enabled: false
      observabilityAddonSpec: {}
      storageConfig:
        metricObjectStorage:
          name: thanos-object-storage
          key: thanos.yaml
    ```

    Note: Deploy a custom image by adding the annotation: `mco-multicluster_observability_addon-image: quay.io/YOUR_ORG_HERE/multicluster-observability-addon:YOUR_TAG_HERE`
```

File: README.md [hunk 2]
```diff
@@ -37,17 +35,63 @@ The logging-ocm-addon consists of one component:
...
 
-## Demo
+1. The addon should now be installed in you hub cluster
+
+```shell
+ kubectl get ClusterManagementAddOn multicluster-observability-addon
+```
```
Comment (xperimental) [inline README.md:87]: Same comment about counting, also removed an extra space:

```suggestion
    ```shell
    kubectl get ClusterManagementAddOn multicluster-observability-addon
    ```
```

**Review by xperimental — COMMENTED** [review]
Gave the Markdown a quick render and had a look, see comments. The comment about enumerations also applies to the "Installing via Kustomize" part, but I can't make suggestions there.

**Review by iblancasa — COMMENTED** [review]
Thanks for removing the demo but I think we should have a way to deploy some resources at least to test what we are developing.
Until now we have been using the demo so...

**Comment by JoaoBraveCoding** [comment]
> Thanks for removing the demo but I think we should have a way to deploy some resources at least to test what we are developing.
Until now we have been using the demo so...

We now have a helm chart under the hack folder that we can use to install an up-to-date version of the add-on. If there are other resources you would like to have under the hack folder, I'm happy to add them either under the addon-install or create a new helm chart for them.

**Comment by iblancasa** [comment]
> > Thanks for removing the demo but I think we should have a way to deploy some resources at least to test what we are developing.
> > Until now we have been using the demo so...
> 
> We now have a helm chart under the hack folder that we can use to install an up-to-date version of the add-on. If there are other resources you would like to have under the hack folder, I'm happy to add them either under the addon-install or create a new helm chart for them.

Could you add those instructions to the `CONTRIBUTING.md` file or something?

**Review by iblancasa — APPROVED** [review]

**Pull request merged**